src/ManifoldsBase.jl

```diff
@@ -144,6 +144,44 @@ level, whether its decorated or not. Any negative value deactivates this depth l
 """
 base_manifold(M::AbstractManifold, ::Val = Val(-1)) = M
 
+
+"""
+    check_approx(M::AbstractManifold, p, q; kwargs...)
+    check_approx(M::AbstractManifold, p, X, Y; kwargs...)
+
+Check whether two elements are approximately equal, either `p`, `q` on the [`AbstractManifold`](@ref)
+or the two tangent vectors `X`, `Y` in the tangent space at `p` are approximately the same.
+The keyword arguments `kwargs` can be used to set tolerances, similar to Julia's `isapprox`.
+
+This function might use `isapprox` from Julia internally and is similar to [`isapprox`](@ref),
+with the difference that is returns an [`ApproximatelyError`](@ref) if the two elements are
+not approximately equal, containting a more detailed description/reason.
+If the two elements are approximalely equal, this method returns `nothing`.
+
+This method is an internal function and is called by `isapprox` whenever the user specifies
+an `error=` keyword therein.
+"""
+function check_approx(M::AbstractManifold, p, q; kwargs...)
+    # fall back to classical approx mode - just that we do not have a reason then
+    res = isapprox(p, q; kwargs...)
```

Comment:
Should it be `isapprox(p, q)` or `isapprox(M, p, q)`?

Reply:
Well, `isapprox(M, p, q)` internally calls `check_approx`, so that would not work.

This call is similar to the default we had before for `isapprox(M,p,q)` which also called this.

Reply:
OK, I see.